packages/blade/src/components/Box/styledProps/getStyledProps.ts

```diff
@@ -48,7 +48,9 @@ const removeUndefinedStyledProps = (obj: StyledPropsInputType): StyledPropsInput
  *
  * **Use `getStyledProps` instead if you're using this for adding styledProps!**
  */
-const makeStyledProps = (props: StyledPropsInputType): KeysRequired<StyledPropsBlade> => {
+const makeStyledProps = (
+  props: StyledPropsInputType,
+): KeysRequired<Omit<StyledPropsBlade, 'pointerEvents' | 'placeItems'>> => {
```

Comment:
```suggestion
): KeysRequired<StyledPropsBlade> => {
```

This is expected to be required. You can add `pointerEvents: props.pointerEvents` and `placeItems: props.placeItems` in the code.

This function basically filters the props so it makes sure that props are being passed to the BaseBox correctly. The placeItems won't work as expected without passing it down from here

Reply:
I didn't add this as per https://github.com/razorpay/blade/pull/1500#discussion_r1282655823

Reply:
Right in that case that `placeItems` value shouldn't exist on StyledPropsBlade type.

StyledPropsBlade is the type of styled-props that we support and `placeItems` and `pointerEvents` are not the styled-props.

Reply:
I didn't get this, so should we use `KeysRequired<StyledPropsBlade>` or `Omit<StyledPropsBlade, 'pointerEvents' | 'placeItems'>>?`

Reply:
We should use `KeysRequired<StyledPropsBlade>` and remove the `placeItems` and `pointerEvents` from actual `StyledPropsBlade` type in basebox types file

Reply:
Basically you're having to omit it because that prop exists in that type right? but that prop shouldn't exist in that type.